app/check-update/page.tsx

```
@@ -66,10 +75,57 @@ export default function CheckUpdatePage() {
+            });
+            return;
+          }
+
+          setInfo({
+            status: "available",
+            version: preparedVersion,
+            changelog: config.changelog ?? "",
+            canCheck: true,
+            lastCheckTime: Math.floor(Date.now() / 1000),
+          });
+          const downloadOk = await handleDownload();
```

ESLint

[error] Error: Cannot access variable before it is declared

`handleDownload` is accessed before it is declared, which prevents the earlier access from updating when this value changes over time.

/home/jailuser/git/app/check-update/page.tsx:110:36
  108 |             lastCheckTime: Math.floor(Date.now() / 1000),
  109 |           });
> 110 |           const downloadOk = await handleDownload();
      |                                    ^^^^^^^^^^^^^^ `handleDownload` accessed before it is declared
  111 |           if (downloadOk) {
  112 |             setTimeout(() => handleInstall(), 600);
  113 |           }

/home/jailuser/git/app/check-update/page.tsx:221:3
  219 |   };
  220 |
> 221 |   const handleDownload = async (): Promise<boolean> => {
      |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 222 |     setDownloadProgress(0);
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 223 |     setInfo((prev) => ({ ...prev, status: "downloading", errorMessage: undefined }));
      …
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 245 |     }
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 246 |   };
      | ^^^^^ `handleDownload` is declared here
  247 |
  248 |   const handleInstall = async (): Promise<boolean> => {
  249 |     setInfo((prev) => ({ ...prev, status: "installing", errorMessage: undefined }));

(react-hooks/immutability)